components/ai_chat/browser/ai_chat_tab_helper.h

```diff
@@ -152,6 +154,8 @@ class AIChatTabHelper : public content::WebContentsObserver,
 
   raw_ptr<AIChatMetrics> ai_chat_metrics_;
 
+  std::deque<mojom::ConversationTurn> pending_requests_;
```

Comment:
That's great, but I do think we only need to consider a single pending request at this time, and just replace any existing pending request if a new one is provided. Especially since it's not clear that the second request is sent with the context of the first, which could be confusing given the conversational nature.  Might also keep the code simpler.

Reply:
I've approved the PR for now without this change as it's not broken, and we can change in a follow-up if we go that way.

Reply:
Cool. changed to have a single pending request.